src/MIDebugEngine/AD7.Impl/AD7MemoryAddress.cs

```diff
@@ -42,6 +39,7 @@ public void SetDocumentContext(IDebugDocumentContext2 docContext)
         // Adds a specified value to the current context's address to create a new context.
         public int Add(ulong dwCount, out IDebugMemoryContext2 newAddress)
         {
+            // FIXME: this is not correct for IDebugCodeContext2
```

Comment:
>// FIXME: this is not correct for IDebugCodeContext2 [](start = 12, length = 52)

Why do you think this is wrong? `dwCount` should be a byte count, not an instruction count.

Reply:
> IDebugCodeContext2 is incremented by instructions rather than by the basic storage size (typically byte).

https://docs.microsoft.com/en-us/visualstudio/extensibility/debugger/reference/idebugcodecontext2#remarks

Which is still not precise enough btw. Could mean Add has pointer semantics so +1 means next instruction or it's just aligned to the next instruction.

Reply:
Got it. That lines speaks to an an idea that I think the original designers of the IDebug* interfaces had. But in practice, I don't think anything in the VS UI will actually try to 'Add' to a code context except for the fact that a code context is also a memory context, so if you wind up getting one in the memory window. In such a memory window situation, the caller would expect Add to move by bytes.

The disassembly window does need to move by instructions, but it does so using [IDebugDisassemblyStream2.Seek](https://docs.microsoft.com/en-us/visualstudio/extensibility/debugger/reference/idebugdisassemblystream2-seek?view=vs-2019).